src/deweb/redirection.ts

```diff
@@ -1,6 +1,33 @@
 export const DEWEB_REDIRECT_URL = 'https://deweb.massa.network/deweb_redirect'
 
-export async function resolveDeweb(uri: string): Promise<string> {
+/* deweb plugin status */
+const STATION_DEWEB_STATUS_URL =
+  'https://station.massa/plugin/massa-labs/local-deweb-provider/api/server/status'
+const DEWEB_STATUS_RUNNING = 'running'
+
+type DewebStatus = {
+  serverPort: number
+  status: string
+}
+
+/**
+ * Resolve the deweb mns domain to a valid http url pointing to a deweb provider.
+ *
+ * Providers are used in following order if availables:
+ * - Check if the mns domain is natively supported by the current browser
+ * - Check if the station deweb plugin is available
+ * - Check if a localhost deweb provider is available
+ * - Check if the current domain is a Deweb provider
+ *
+ * If no provider is available, we redirect to the default URL
+ * @param uri - The mns domain to resolve. It must be suffixed with a '.massa'
+ * @param chainId - The returned url must belong to a provider having the same same chainid. If null, we don't check the chainId
+ * @returns The resolved URI
+ */
+export async function resolveDeweb(
+  uri: string,
+  chainId: number | null = null
```

Comment:
in ts, always prefer the use of `undefined` over `null`
here just use `chainId?: bigint`
Use bigint to keep coherency with massa-web3